spring-web-modules/spring-mvc-file/src/main/java/com/baeldung/file/CustomMultipartFile.java

```diff
@@ -0,0 +1,58 @@
+package com.baeldung.file;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.springframework.web.multipart.MultipartFile;
+
+public class CustomMultipartFile implements MultipartFile {
+    private byte[] input;
+
+    public CustomMultipartFile(byte[] input) {
+        this.input = input;
+    }
+
+    @Override
+    public String getName() {
+        return null;
+    }
+
+    @Override
+    public String getOriginalFilename() {
+        return null;
+    }
+
+    @Override
+    public String getContentType() {
+        return null;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return input == null || input.length == 0;
+    }
+
+    @Override
+    public long getSize() {
+        return input.length;
+    }
+
+    @Override
+    public byte[] getBytes() throws IOException {
+        return input;
+    }
+
+    @Override
+    public InputStream getInputStream() throws IOException {
+        return new ByteArrayInputStream(input);
+    }
+
+    @Override
+    public void transferTo(File destination) throws IOException, IllegalStateException {
+        new FileOutputStream(destination).write(input);
+    }
```

Comment:
We must make sure that the stream is close, so the best way for an Autoclosable is using try-with-resources pattern. So this should be refactored to something like:
```
        try(FileOutputStream fos = new FileOutputStream(destination)) {
            fos.write(input);
        }
```